src/main/scala/com/microsoft/hyperspace/index/sources/default/DefaultFileBasedSource.scala

```diff
@@ -67,8 +67,19 @@ class DefaultFileBasedSource(private val spark: SparkSession) extends FileBasedS
         val sourceDataProperties =
           Hdfs.Properties(Content.fromLeafFiles(files, fileIdTracker).get)
         val fileFormatName = fileFormat.asInstanceOf[DataSourceRegister].shortName
+
+        // Store basePath of hive-partitioned data sources, if applicable.
+        val basePath = PathUtils.extractBasePath(location.partitionSpec())
```

Comment:
Could you use basePath if it's given? as it's already validated.

Reply:
thanks @sezruby , added